kpi-tracker/src/main.rs

```diff
@@ -749,43 +759,55 @@ async fn process_chain_genesis_block(
     Ok(genesis_data)
 }
 
+#[derive(Debug, Clone, Copy)]
+struct PaydayBlockData {
+    total_stake: Amount,
+    /// The amount of active bakers. Only for >= protocol version 6.
+    baker_count: Option<i64>,
+}
+
 /// If block specified by `block_hash` is a payday block (also implies >=
-/// protocol version 4), this returns the total stake for that block. Otherwise
-/// returns `None`.
-async fn p4_payday_total_stake(
+/// protocol version 4), this returns [`PaydayBlockData`]. Otherwise, returns
+/// `None`.
+async fn process_payday_block(
     node: &mut Client,
     block_hash: BlockHash,
-) -> anyhow::Result<Option<Amount>> {
+) -> anyhow::Result<Option<PaydayBlockData>> {
+    let is_payday_block = node
+        .is_payday_block(block_hash)
+        .await
+        .with_context(|| {
+            format!("Could not assert whether block is payday block for: {block_hash}")
+        })?
+        .response;
+
+    if !is_payday_block {
+        return Ok(None);
+    }
+
     let tokenomics_info = node
         .get_tokenomics_info(block_hash)
         .await
-        .with_context(|| format!("Could not get tokenomics info for block: {}", block_hash))?
+        .with_context(|| format!("Could not get tokenomics info for block: {block_hash}"))?
         .response;
 
-    if let RewardsOverview::V1 {
+    let RewardsOverview::V1 {
         total_staked_capital,
         ..
-    } = tokenomics_info
-    {
-        let is_payday_block = node
-            .is_payday_block(block_hash)
-            .await
-            .with_context(|| {
-                format!(
-                    "Could not assert whether block is payday block for: {}",
-                    block_hash
-                )
-            })?
-            .response;
-
-        if is_payday_block {
-            return Ok(Some(total_staked_capital));
-        };
-
+    } = tokenomics_info else {
         return Ok(None);
-    }
+    };
+
+    let baker_count = match node.get_bakers_reward_period(block_hash).await {
```

Comment:
This is a bit ambiguous. This will compute the number of bakers in a reward period that is just starting, but the payday block distributes rewards for the **previous rewards period**.

Can you clarify what the intended semantics is?